test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchiveUpgrade.java

```diff
@@ -26,19 +26,29 @@
 /*
  * @test
  * @summary Check that -XX:+AutoCreateSharedArchive automatically recreates an archive when you change the JDK version.
- * @requires os.family == "linux" & vm.bits == "64" & (os.arch=="amd64" | os.arch=="x86_64")
+ * @requires vm.cds & vm.bits == "64"
  * @library /test/lib
  * @compile -source 1.8 -target 1.8 ../test-classes/HelloJDK8.java
  * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar Hello.jar HelloJDK8
  * @run driver TestAutoCreateSharedArchiveUpgrade
  */
 
 import java.io.File;
+import java.util.HashMap;
+import java.util.Properties;
+import jdk.test.lib.artifacts.Artifact;
+import jdk.test.lib.artifacts.ArtifactResolver;
+import jdk.test.lib.artifacts.ArtifactResolverException;
```

Comment:
Some of these new imports are not needed anymore. Also, there's other code that need to be removed, like the `props` below.